packages/block-library/src/image/index.js

```diff
@@ -186,7 +186,7 @@ export const settings = {
 					caption: {
 						shortcode: ( attributes, { shortcode } ) => {
 							const { content } = shortcode;
-							return content.replace( /\s*<img[^>]*>\s/, '' );
+							return content.replace( /\s*(?:<a[^>]*>\s*)?<img[^>]*>(?:\s*<\/a>)?\s*/, '' );
```

Comment:
Could we handle this with more confidence by natively parsing HTML, instead of using RegExp?

Reply:
What's our target browsers? We can use the browser's dom parser to parse
```
if ( typeof DOMParser !== 'undefined' && DOMParser.prototype.parseFromString ) {
	const parser = new DOMParser();
	const parsed = parser.parseFromString( html, 'text/html' );
}
```

Reply:
Ah https://developer.mozilla.org/en-US/docs/Web/API/DOMParser has an example of a try catch fallback to regex, if it's not available.

Reply:
noting that all of the supported browsers in `@wordpress/browserslist-config` provide the `DOMParser()`

Reply:
Spinning up an example alternative

Reply:
Oh sorry, didn't see the last comments.